domain/src/main/kotlin/no/nav/su/se/bakover/domain/vedtak/Vedtak.kt

```diff
@@ -1,24 +1,36 @@
 package no.nav.su.se.bakover.domain.vedtak
 
+import arrow.core.Either
 import no.nav.su.se.bakover.common.Tidspunkt
 import no.nav.su.se.bakover.common.UUID30
 import no.nav.su.se.bakover.common.periode.Periode
 import no.nav.su.se.bakover.domain.NavIdentBruker
+import no.nav.su.se.bakover.domain.behandling.AvslagGrunnetBeregning
 import no.nav.su.se.bakover.domain.behandling.Behandling
 import no.nav.su.se.bakover.domain.behandling.Behandlingsinformasjon
+import no.nav.su.se.bakover.domain.behandling.VurderAvslagGrunnetBeregning
+import no.nav.su.se.bakover.domain.behandling.avslag.Avslagsgrunn
 import no.nav.su.se.bakover.domain.beregning.Beregning
-import no.nav.su.se.bakover.domain.eksterneiverksettingssteg.EksterneIverksettingsstegEtterUtbetaling
-import no.nav.su.se.bakover.domain.eksterneiverksettingssteg.EksterneIverksettingsstegForAvslag
+import no.nav.su.se.bakover.domain.brev.BrevbestillingId
+import no.nav.su.se.bakover.domain.eksterneiverksettingssteg.JournalføringOgBrevdistribusjon
+import no.nav.su.se.bakover.domain.eksterneiverksettingssteg.JournalføringOgBrevdistribusjonFeil
+import no.nav.su.se.bakover.domain.journal.JournalpostId
 import no.nav.su.se.bakover.domain.oppdrag.simulering.Simulering
 import no.nav.su.se.bakover.domain.revurdering.IverksattRevurdering
+import no.nav.su.se.bakover.domain.søknadsbehandling.ErAvslag
 import no.nav.su.se.bakover.domain.søknadsbehandling.Søknadsbehandling
+import no.nav.su.se.bakover.domain.visitor.Visitable
 import java.util.UUID
 
-sealed class Vedtak {
+sealed class Vedtak : Visitable<VedtakVisitor> {
```

Comment:
Jeg må innrømme at en liten del av meg dør hver gang det kommer en ny visitor!

Reply:
Hehe, jeg vet. 

Først og fremst var motivasjonen min å sørge for at de samme brevene lages likt hver gang. All den tid vi oppretter "det samme" brevet ad-hoc på mange forskjellige steder, føles det litt tryggere å vite at dette går igjennom samme algoritmen hver gang, heller enn at det er en ad-hoc request opprettes "der og da". Skulle noe være vise seg å være feil, kan man muligens også slippe unna med å fikse det der, heller enn å måtte innom 6 steder hvor "den samme" requesten lages ad-hoc.